google-http-client/src/main/java/com/google/api/client/http/HttpResponseException.java

```diff
@@ -253,31 +293,45 @@ public HttpHeaders getHeaders() {
     /**
      * Sets the HTTP response headers.
      *
-     * <p>
-     * Overriding is only supported for the purpose of calling the super implementation and changing
-     * the return type, but nothing else.
-     * </p>
+     * <p>Overriding is only supported for the purpose of calling the super implementation and
+     * changing the return type, but nothing else.
      */
     public Builder setHeaders(HttpHeaders headers) {
       this.headers = Preconditions.checkNotNull(headers);
       return this;
     }
 
     /** Returns the HTTP response content or {@code null} for none. */
-    public final String getContent() {
-      return content;
+    public final String getContent() throws IOException {
+      if (content == null) {
+        return null;
+      }
+      return new String(content, charset.name());
     }
 
     /**
      * Sets the HTTP response content or {@code null} for none.
      *
-     * <p>
-     * Overriding is only supported for the purpose of calling the super implementation and changing
-     * the return type, but nothing else.
-     * </p>
+     * <p>Overriding is only supported for the purpose of calling the super implementation and
+     * changing the return type, but nothing else.
      */
     public Builder setContent(String content) {
-      this.content = content;
+      if (content != null) {
+        this.content = content.getBytes();
```

Comment:
Should this set the content to null or perhaps empty if null?

Reply:
@elharo as a class field byte[] content; is null by initialization, I would assume that to make it empty when the string is null make no sense because the string could be empty.